lib/node_modules/@stdlib/repl/lib/commands/settings.js

```diff
@@ -0,0 +1,136 @@
+var isBoolean = require( '@stdlib/assert/is-boolean' );
+var hasOwnProp = require( '@stdlib/assert/has-own-property' );
+var objectKeys = require( '@stdlib/utils/keys' );
+var format = require( '@stdlib/string/format' );
+
+
+// VARIABLES //
+
+var debug = logger( 'repl:command:settings' );
+var SETTINGS = {
+	'autoCloseSymbols': {
+		'desc': 'Auto-close brackets and quotations.'
+	}
+};
+
+
+// FUNCTIONS //
+
+/**
+* Returns settings help text.
+*
+* @private
+* @returns {string} settings help text
+*/
+function help() {
+	var HELP_TEXT;
+	var names;
+	var o;
+	var i;
+
+	names = objectKeys( SETTINGS );
+	HELP_TEXT = '\n';
+	for ( i = 0; i < names.length; i++ ) {
+		o = SETTINGS[ names[ i ] ];
+		HELP_TEXT += names[ i ] + '\n';
+		HELP_TEXT += '    ';
+		if ( o.desc ) {
+			HELP_TEXT += SETTINGS[ names[ i ] ].desc;
+		} else {
+			HELP_TEXT += '(no description available)';
+		}
+		HELP_TEXT += '\n\n';
+	}
+	return HELP_TEXT;
+}
+
+
+// MAIN //
+
+/**
+* Returns a callback to be invoked upon calling the `settings` command.
+*
+* @private
+* @param {REPL} repl - REPL instance
+* @returns {Function} callback
+*/
+function command( repl ) {
+	return onCommand;
+
+	/**
+	 * Manages REPL settings.
+	 *
+	 * If no arguments are given, it prints all settings with their description.
+	 * If one argument is given, it fetches the value of the setting with the given name and prints it.
+	 * If two arguments are given, it updates the value of the given setting name to the given value.
+	 *
+	 * @private
+	 * @param {string} [name] - Setting name
+	 * @param {boolean} [value] - New setting value
+	 * @throws {TypeError} First argument must be a string
+	 * @throws {Error} Must recognize setting name
+	 * @throws {TypeError} Second argument must be a boolean
+	 */
+	function onCommand( name, value ) {
+		var err;
+		if ( arguments.length === 0 ) {
+			repl._ostream.write( help() );
+			return;
+		}
+		if ( !isString( name ) ) {
+			err = new TypeError( format( 'invalid argument. First argument must be a string. Value: `%s`.', name ) );
+			debug( 'Error: %s', err.message );
+			repl._ostream.write( format( 'Error: %s\n.', err.message ) );
+			return;
+		}
+		if ( !hasOwnProp( SETTINGS, name ) ) {
+			err = new Error( format( 'invalid argument. Unrecognized setting. Value: `%s`.', name ) );
+			debug( 'Error: %s', err.message );
+			repl._ostream.write( format( 'Error: %s\n.', err.message ) );
+			return;
+		}
+		if (arguments.length === 1) {
+			repl._ostream.write('\n'+repl._settings[name]+'\n\n');
+			return;
+		}
+		if ( !isBoolean(value) ) {
+			err = new TypeError( format( 'invalid argument. Second argument must be a boolean. Value: `%s`.', name ) );
+			debug( 'Error: %s', err.message );
+			repl._ostream.write( format( 'Error: %s\n.', err.message ) );
+			return;
+		}
+		repl._settings[name] = value;
+		repl._ostream.write( format( '\nUpdated setting %s to %s.\n\n', name, ( value ) ? 'true' : 'false' ) );
```

Comment:
I don't think you need the ternary here (a) and (b) again, we should not assume that all settings will be booleans.